rplidar.go

```diff
@@ -113,30 +134,28 @@ func (rp *Rplidar) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, e
 
 // start requests that the rplidar starts up and starts spinning.
 func (rp *Rplidar) start() {
-	rp.mu.Lock()
```

Comment:
[note] mutex removed from `start`. `start` only gets called in two places:
1. During startup
2. During NextPointCloud if the lidar has not already been started (THIS WOULD DEADLOCK)

Reply:
`start` had a similar problem as Stop but harder to hit. NextPointCloud also acquires the same mutex as `start`.

We are moving taking the lock up a level so that it's taken by the callers of `start`.

I think instead we should just error if GetPointCloud is called before `start` has completed.